code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -332,7 +334,7 @@ type Info struct {
 
 // StorageNode represents Blobber configurations.
 type StorageNode struct {
-	ID                      string                 `json:"id"`
+	*provider.Provider      `json:"provider"`
```

Comment:
We don't have to use pointer embed because the provider field must not be nil, also use value embed could avoid the field changed mistakenly by reference. Same as other providers.

Reply:
Using value embed will mean that any future `Provider` methods will likely need to be defined on `Provider` rather than `*Provider`, or we will need to be systematic about always using pointer objects for handling providers. Although using value embed might be the better evil?